src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/MSBuildShellOutNuGetProject.cs

```diff
@@ -326,6 +355,51 @@ public static string GetMSBuildPath(IVsBuildPropertyStorage buildPropertyStorage
             return null;
         }
 
+        public bool IsRestoreRequired(
+            IEnumerable<VersionFolderPathResolver> pathResolvers,
+            ISet<PackageIdentity> packagesChecked,
+            ExternalProjectReferenceContext context)
+        {
+            return true;
+        }
+
+        public async Task<IReadOnlyList<ExternalProjectReference>> GetProjectReferenceClosureAsync(
+            ExternalProjectReferenceContext context)
+        {
+            await TaskScheduler.Default;
```

Comment:
why don't we update the context cache in this case? else what is the use of ExternalProjectReferenceContext  here?